frontend/src/components/pages/AdminDashboard/EditCheckInOrFoodRescue.tsx

```diff
@@ -0,0 +1,125 @@
+import {
+  Box,
+  CloseButton,
+  Container,
+  Input,
+  Text,
+  Textarea,
+} from "@chakra-ui/react";
+import React, { useEffect, useState } from "react";
+import { useForm } from "react-hooks-helper";
+import { useHistory } from "react-router-dom";
+
+import ContentAPIClient from "../../../APIClients/ContentAPIClient";
+import * as Routes from "../../../constants/Routes";
+import { Content } from "../../../types/ContentTypes";
+import SaveButton from "../Scheduling/SaveChangesButton";
+
+const EditCheckInOrFoodRescue = ({
+  isCheckInView = false,
+}: {
+  isCheckInView?: boolean;
+}): React.ReactElement => {
+  const [content, setContent] = useState<Content>();
+  const [{ description, url }, setValue] = useForm({
+    description: "",
+    url: "",
+  });
+
+  const history = useHistory();
+  const navigateToViewPage = () => {
+    history.push(
+      isCheckInView ? Routes.ADMIN_CHECK_INS : Routes.ADMIN_VIEW_DONATIONS,
+    );
+  };
+
+  useEffect(() => {
+    const getContent = async () => {
+      const contentResponse = await ContentAPIClient.getContent();
+      setContent(contentResponse);
+    };
+
+    getContent();
+  }, []);
+
+  const onSaveClick = async () => {
+    const contentResponse = await ContentAPIClient.updateContent(
+      content?.id ?? "",
+      {
+        checkinDescription:
+          (isCheckInView ? description : content?.checkinDescription) ?? "",
+        checkinUrl: (isCheckInView ? url : content?.checkinUrl) ?? "",
+        foodRescueDescription:
+          (!isCheckInView ? description : content?.foodRescueDescription) ?? "",
+        foodRescueUrl: (!isCheckInView ? url : content?.foodRescueUrl) ?? "",
```

Comment:
I realize we actually shouldn't allow these values to be empty, they should always have content. This means that we should validate for if a field is empty. I guess there aren't error messages in Figma but we can just highlight the box in red.

You can do `isInvalid={!description && interactedWith}` inside of the `Textarea/>` component for the description and url
with `const [interactedWith, setInteractedWith] = useState(false);` for if you have interacted with the text box (so it's not red when you first enter the page)

that should let you get rid of this nullish coalescing happening